.github/workflows/ci.yml

```diff
@@ -578,7 +578,7 @@ jobs:
             bash -euo pipefail -c \
             "./build_tools/cmake/build_riscv.sh && tests/riscv32/smoke.sh"
```

Comment:
Maybe we can put `tests/riscv32/smoke.sh` into `./build_tools/cmake/test_riscv.sh`, so this test can be part of the matrix too. I guess we can simply invoke smoke.sh from test_riscv.sh when BUILD_ARCH=rv32-baremetal.

Reply:
I want to separate baremetal vs linux, especially when https://github.com/iree-org/iree/pull/10442 is landed and we will set the `BUILD_PRESET=benchmark-suite-test` for the linux targets. Making them all part of the matrix may hurt the readability.